osu.Game/GameModes/Play/BeatmapGroup.cs

```diff
@@ -0,0 +1,246 @@
+            set
+            {
+                if (collapsed == value)
+                    return;
+                collapsed = value;
+                this.ClearTransformations();
+                const float uncollapsedAlpha = 1;
+                Transforms.Add(new TransformAlpha(Clock)
+                {
+                    StartValue = collapsed ? uncollapsedAlpha : collapsedAlpha,
+                    EndValue = collapsed ? collapsedAlpha : uncollapsedAlpha,
+                    StartTime = Time,
+                    EndTime = Time + 250,
+                });
+                if (collapsed)
+                    topContainer.Remove(difficulties);
+                else
+                    topContainer.Add(difficulties);
+                setBox.ClearTransformations();
+                setBox.Width = collapsed ? collapsedWidth : 1; // TODO: Transform
+                setBox.BorderColour = new Color4(
+                    setBox.BorderColour.R,
+                    setBox.BorderColour.G,
+                    setBox.BorderColour.B,
+                    collapsed ? 0 : 255);
+                setBox.GlowRadius = collapsed ? 0 : 5;
+            }
+        }
+        
+        private void updateSelected(BeatmapInfo map)
+        {
+            int selected = BeatmapSet.Beatmaps.IndexOf(map);
+            var buttons = difficulties.Children.ToList();
+            for (int i = 0; i < buttons.Count; i++)
+            {
+                var button = buttons[i] as BeatmapButton;
+                float targetWidth = 1 - Math.Abs((selected - i) * 0.025f);
+                targetWidth = MathHelper.Clamp(targetWidth, 0.8f, 1);
+                button.Width = targetWidth; // TODO: Transform
+                button.Selected = selected == i;
+            }
+            BeatmapSelected?.Invoke(BeatmapSet, map);
+        }
+
+        public BeatmapGroup(BeatmapSetInfo beatmapSet)
+        {
+            BeatmapSet = beatmapSet;
+            selectedBeatmap = beatmapSet.Beatmaps[0];
+            Alpha = collapsedAlpha;
+            AutoSizeAxes = Axes.Y;
+            RelativeSizeAxes = Axes.X;
+            float difficultyWidth = 1;
+            Children = new[]
+            {
+                topContainer = new FlowContainer
+                {
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
+                    Direction = FlowDirection.VerticalOnly,
+                    Children = new[]
+                    {
+                        setBox = new BeatmapSetBox(beatmapSet)
+                        {
+                            RelativeSizeAxes = Axes.X,
+                            Width = collapsedWidth,
+                            Anchor = Anchor.TopRight,
+                            Origin = Anchor.TopRight,
+                        }
+                    }
+                }
+            };
+            difficulties = new FlowContainer // Deliberately not added to children
```

Comment:
i'd just add this from the start and make it invisible. no real overhead from doing so.